### PR TITLE
kernel: allow root processes to get version

### PR DESCRIPTION
**Pull request by Scirese**

### CODE REVIEW

File: kernel/ksu.c [hunk 1]
```diff
@@ -244,6 +244,16 @@ static int handler_pre(struct kprobe *p, struct pt_regs *regs)
 		return 0;
 	}
 
+	// Both root manager and root processes should be allowed to get version
+	if( __manager_uid == current_uid().val || 0 == current_uid().val ) {
```
Comment (tiann) [inline kernel/ksu.c:248]: Check the arg2 first, and then check if it is valid role.
And also, please reuse the `is_manager`, don't repeat yourself.

File: kernel/ksu.c [hunk 2]
```diff
@@ -244,6 +244,16 @@ static int handler_pre(struct kprobe *p, struct pt_regs *regs)
...
+		if (arg2 == CMD_GET_VERSION) {
+			u32 version = KERNEL_SU_VERSION;
+			if (copy_to_user(arg3, &version, sizeof(version))) {
+				pr_err("prctl reply error, cmd: %d\n", arg2);
+			}
```
Comment (tiann) [inline kernel/ksu.c:253]: we should return here.

**Review by tiann — CHANGES_REQUESTED** [review]

**Comment by Scirese** [comment]
```	
// Both root manager and root processes should be allowed to get version
	if (arg2 == CMD_GET_VERSION) {
		if( is_manager() || 0 == current_uid().val ) {
			u32 version = KERNEL_SU_VERSION;
			if (copy_to_user(arg3, &version, sizeof(version))) {
				pr_err("prctl reply error, cmd: %d\n", arg2);
				return 0;
			}
		}
	}
```
Like this?

**Comment by tiann** [comment]
> ```
> // Both root manager and root processes should be allowed to get version
> 	if (arg2 == CMD_GET_VERSION) {
> 		if( is_manager() || 0 == current_uid().val ) {
> 			u32 version = KERNEL_SU_VERSION;
> 			if (copy_to_user(arg3, &version, sizeof(version))) {
> 				pr_err("prctl reply error, cmd: %d\n", arg2);
> 				return 0;
> 			}
> 		}
> 	}
> ```
> 
> Like this? ~(let me sure before pushing commits)~

looks good

**Comment by tiann** [comment]
But the second if should be formatted.

**Comment by Scirese** [comment]
OK

File: kernel/ksu.c [hunk 3]
```diff
@@ -244,6 +244,17 @@ static int handler_pre(struct kprobe *p, struct pt_regs *regs)
 		return 0;
 	}
 
+	// Both root manager and root processes should be allowed to get version
+	if (arg2 == CMD_GET_VERSION) {
+		if( is_manager() || 0 == current_uid().val ) {
+			u32 version = KERNEL_SU_VERSION;
+				if (copy_to_user(arg3, &version, sizeof(version))) {
```
Comment (Ylarod) [inline kernel/ksu.c:251]: Incorrect indent

**Review by Ylarod — CHANGES_REQUESTED** [review]

File: kernel/ksu.c (hunk 3 above)
Comment (Scirese) [reply kernel/ksu.c:251]: I don't want CI to rerun, could you please fix it after merge?

Comment (tiann) [reply kernel/ksu.c]: It is ok to let it run

Comment (Ylarod) [reply kernel/ksu.c:251]: add `[skip ci]` to your commit message
https://docs.github.com/zh/actions/managing-workflow-runs/skipping-workflow-runs

**Review by Ylarod — APPROVED** [review]

**Review by tiann — APPROVED** [review]

**Pull request merged**